app/shared/reducers/validate.js

```diff
@@ -10,14 +10,36 @@ export default function validate(state = {}, action) {
     return {};
   }
 
-  const matches = /^VALIDATE_(.*)_(PENDING|SUCCESS|FAILURE|CONFIRMING|NULL)$/.exec(type);
+  const matches = /^VALIDATE_(.*)_(PENDING|SUCCESS|FAILURE|NULL)$/.exec(type);
   if (!matches) return state;
 
   const [, requestName, requestState] = matches;
 
-  return {
+  const errField = `${requestName}_LAST_ERROR`;
+  const txField = `${requestName}_LAST_TRANSACTION`;
+
+  const newState = {
     ...state,
     [requestName]: requestState,
-    [`${requestName}_ERROR`]: action.payload && action.payload.error
+    [errField]: '',
+    [txField]: ''
   };
+
+  if (action.payload) {
+    // Attempt to process any errors returned
+
+    if (action.payload.err) {
+      try {
+        newState[errField] = JSON.parse(action.payload.err);
+      } catch (e) {
+        newState[errField] = action.payload.err;
+      }
+    }
+
+    // Attach any returned transactions
+    if (action.payload.tx) {
+      newState[txField] = action.payload.tx;
+    }
+  }
+  return newState;
```

Comment:
The `txField` and `errField` should be handled in the `system` actions.

Reply:
ok if that's ok with you, I'll create a system/stake file and move that there.